### PR TITLE
Implement logout routing

### PR DESCRIPTION
**Pull request by RtAnto**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.
- [x ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

tested routes using postman with success

# Notes

may need to update the database functions to remove sessions to fix the case where a user logouts from sso with multiple sessions active

# Pivotal Tracker Link:

PT link here

### CODE REVIEW

File: Backend/DataAccessLayer/Repositories/SessionRepository.cs [hunk 1]
```diff
@@ -18,7 +18,15 @@ public Session GetSession(DatabaseContext _db, string token)
 
             return session;
         }
-        
+
+        public Session GetSession(DatabaseContext _db, User user)
```
Comment (Luke-Clements) [inline Backend/DataAccessLayer/Repositories/SessionRepository.cs:22]: Sessions shouldn't be looked up by user, only by token.

File: Backend/DataAccessLayer/Repositories/SessionRepository.cs [hunk 2]
```diff
@@ -63,5 +71,12 @@ public Session ExpireSession(DatabaseContext _db, string token)
             _db.Entry(session).State = EntityState.Modified;
             return session;
         }
+
+        public Guid GetSessionGuid(DatabaseContext _db, string token)
```
Comment (Luke-Clements) [inline Backend/DataAccessLayer/Repositories/SessionRepository.cs:75]: No need to retrieve the Session Guid with a separate function. It is already accessible in the object.

**Comment by Luke-Clements** [comment]
A lot of this should be in a different pr (Vue stuff). Master needs to be merged into this branch before it can be merged.

**Comment by RtAnto** [comment]
Branched off of alfredo's brnach, this is going to be nasty so i'm going to redo submit another pr

**Pull request closed without merging**